sdk/src/test/java/com/silanis/esl/sdk/internal/converter/utils/ReferencedConditionsConverterTestUtil.java

```diff
@@ -0,0 +1,170 @@
+        if (apiRefDocs.size() != sdkRefDocs.size()) {
+            return false;
+        }
+
+        for (int i = 0; i < apiRefDocs.size(); i++) {
+            if (!compareReferencedDocuments(apiRefDocs.get(i), sdkRefDocs.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean compareReferencedFields(List<com.silanis.esl.api.model.ReferencedField> apiRefFields, List<ReferencedField> sdkRefFields) {
+        if (apiRefFields.size() != sdkRefFields.size()) {
+            return false;
+        }
+
+        for (int i = 0; i < apiRefFields.size(); i++) {
+            if (!compareReferencedFields(apiRefFields.get(i), sdkRefFields.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static boolean compareReferencedFields(com.silanis.esl.api.model.ReferencedField apiRefField, ReferencedField sdkRefField) {
+        return apiRefField.getFieldId().equals(sdkRefField.getFieldId()) &&
+            compareReferencedFieldConditions(apiRefField.getConditions(), sdkRefField.getConditions());
+    }
+
+    public static boolean compareReferencedFieldConditions(com.silanis.esl.api.model.ReferencedFieldConditions apiReferencedFieldConditions,
+        ReferencedFieldConditions sdkReferencedFieldConditions) {
+        List<com.silanis.esl.api.model.FieldCondition> apiReferencedInCondition = apiReferencedFieldConditions.getReferencedInCondition();
+        List<com.silanis.esl.api.model.FieldCondition> apiReferencedInAction = apiReferencedFieldConditions.getReferencedInAction();
+        List<FieldCondition> sdkReferencedInCondition = sdkReferencedFieldConditions.getReferencedInCondition();
+        List<FieldCondition> sdkReferencedInAction = sdkReferencedFieldConditions.getReferencedInAction();
+
+        return compareFieldConditions(apiReferencedInCondition, sdkReferencedInCondition) &&
+            compareFieldConditions(apiReferencedInAction, sdkReferencedInAction);
+    }
+
+    public static boolean compareFieldConditions(com.silanis.esl.api.model.FieldCondition apiFieldCondition, FieldCondition sdkFieldCondition) {
+        return apiFieldCondition.getId().equals(sdkFieldCondition.getId()) &&
+            apiFieldCondition.getCondition().equals(sdkFieldCondition.getCondition()) &&
+            apiFieldCondition.getAction().equals(sdkFieldCondition.getAction());
+    }
+
+    public static boolean compareFieldConditions(List<com.silanis.esl.api.model.FieldCondition> apiFieldConditions, List<FieldCondition> sdkFieldConditions) {
+        if (apiFieldConditions.size() != sdkFieldConditions.size()) {
+            return false;
+        }
+
+        for (int i = 0; i < apiFieldConditions.size(); i++) {
+            if (!compareFieldConditions(apiFieldConditions.get(i), sdkFieldConditions.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+}
```

Comment:
Remove ReferencedConditionsConverterTestUtil. If you want to avoid repetition of the same code, redefine ReferencedConditionsConverterTest as a superclass and inherit from the child.